bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/RequestValidationService.kt

```diff
@@ -40,56 +38,76 @@ class RequestValidationService(
     private val addressRepository: LogisticAddressRepository,
     private val metadataService: MetadataService
 ) {
-    fun validateLegalEntityCreates(
-        requests: Collection<LegalEntityPartnerCreateRequest>
-    ): Map<LegalEntityPartnerCreateRequest, Collection<ErrorInfo<LegalEntityCreateError>>> {
 
-        val legalEntityRequests = requests.map { it.legalEntity }
-        val legalAddressRequests = requests.map { it.legalAddress }
+    fun <IRequest>validateLegalEntityCreates(
+        legalEntitiesByRequest: Map<IRequest, IBaseLegalEntityDto>, entityKeyFunc : (IRequest) -> String?
+    ): Map<IRequest, List<ErrorInfo<LegalEntityCreateError>>> {
```

Comment:
This is quite a complicated function call for being used by other services. Maybe make it private and add public functions that just accept the actual DTOs of Pool and Orchestrator? Then the invoking services have an easier time using this service

Reply:
The split is  definitely important. One Method should do on task. The logic for addresses and legal entities are completely independent from each other.
The validation logic is complicated, so i think should be reused. If we copy the code at the first level it would be easier to understand.

Reply:
We now have these 2 very related validation methods `validateLegalEntityCreates`, `validateLegalEntityCreatesAddresses`. First one is for legal entities, second for legal addresses beneath legal entities.
As far as I can see from a consumer side these two methods are always used together and the results are joined together (in FetchAndReserveCleaningStepService and BusinessPartnerBuildService). So I still think a single *public* method would be easier to use. But if it makes sense to have two separate methods because the logic is independent, you could make them private.